govcd/tm_edge_cluster.go

```diff
@@ -0,0 +1,172 @@
+package govcd
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/vmware/go-vcloud-director/v3/types/v56"
+)
+
+const labelTmEdgeCluster = "TM Edge Cluster"
+const labelTmEdgeClusterSync = "TM Edge Cluster Sync"
+const labelTmEdgeClusterTransportNodeStatus = "TM Edge Cluster Transport Node Status"
+
+// TmEdgeCluster manages read operations for NSX-T Edge Clusters and their QoS settings
+type TmEdgeCluster struct {
+	TmEdgeCluster *types.TmEdgeCluster
+	vcdClient     *VCDClient
+}
+
+// wrap is a hidden helper that facilitates the usage of a generic CRUD function
```

Comment:
This is a bit confusing. By "hidden" you mean private? Do we really need this function? It seems so small and I can't seem to find its usage in this PR.

Reply:
It is everywhere, where we use generic methods. It is required to make generic approach work - https://github.com/vmware/go-vcloud-director/blob/main/govcd/openapi_generic_outer_entities.go#L13-L19

Reply:
Ah, truly. Thanks! Friday before holidays effect..